lua/neo-tree/sources/filesystem/lib/fs_scan.lua

```diff
@@ -206,7 +206,11 @@ local function get_children_sync(path)
 end
 
 local function get_children_async(path, callback)
-  uv.fs_opendir(path, function(_, dir)
+  uv.fs_opendir(path, function(err, dir)
+    if err then
```

Comment:
It's not correct to just silently disregard every error. It should be specific about what errors can be ignored. In all cases, the error should be logged at some level. Probably `debug` for ignored errors and let it crash for other errors.

Reply:
Understood, just pushed a few commits that log permission errors to `debug` and lets the other errors crash.

Also, I see that `get_children_sync` logs to `error` - would that be preferable for the other errors?

Reply:
Now that you mention it, they should be the same. How about they both check for and ignore permission based errors and log an error for anything else.

Reply:
Cool, I made them handle errors the same now and added a check for the Unix permission-based error. This PR now resolves any blocking error spam when encountering permission problems on my Linux and Windows machines in both sync and async modes.

I also removed the `pcall` surrounding the libuv calls, because those libuv functions return `nil, err, name` (the latter 2 are strings) when it errors anyways, so we can check for those instead of pcall's success value if needed.